include/fix8/sessionwrapper.hpp

```diff
@@ -646,10 +646,39 @@ class SessionInstance : public SessionInstanceBase
 	/// Dtor.
 	virtual ~SessionInstance ()
 	{
-		delete _session;
-		_session = nullptr;
-		delete _sock;
-		_sock = nullptr;
+		try
+		{
+			if (_psc != nullptr)
+			{
+				_psc->stop();
+				delete _psc;
+				_psc = nullptr;
+			}
+		}
+		catch (...)
```

Comment:
An exception catch-all is not suitable here. You need to at least specify the base class of the exception you are expecting to catch.